packages/framework/ai-collab/src/shared-tree-diff/sharedTreeBranchManager.ts

```diff
@@ -183,7 +184,10 @@ export class SharedTreeBranchManager {
 			switch (diff.type) {
 				case "CHANGE":
 				case "CREATE": {
-					targetObject.set(diff.path[diff.path.length - 1] as string, diff.value);
+					// This code is doing non-schema aware editing, which is not a supported feature of this API.
```

Comment:
Should we leave a TODO to not do this? Or is this intentional and desired? Not clear to me from the docs.

Reply:
I don't have a lot of context on this ai-collab code. My understanding was that we don't support non-schema aware editing, but this code is doing exactly that. We should probably do one of:
1. support such a use case
2. stop depending on such usage
3. decide this code is a kludge

This package was added after I created this PR, and does not compile with my changes, so I looked at why and found it doing things we do not support which are not correctly blocked from compiling.

I this package is believe based on FHL work. Someone who owns this is going to need to provide some guidance here.

Reply:
Decent chance it's #3. Not sure if the way we're headed (what we're calling the "explicit approach") we'll still need this code. @seanimam might be able to comment. If we keep the code, I suspect we'll want to figure out how to make this schema-aware.